src/pages/AdminPages/UnapprovedAccounts/SelectedUserModal.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react'
+import { useSelector, useDispatch, shallowEqual } from 'react-redux'
+import BaseModal from '../../../components/Modal/BaseModal'
+import * as Yup from 'yup'
+import { Form, FormField, FormSelect, SubmitButton } from '../../../components/Form'
+import { superAdminSettableRoles, adminSettableRoles } from '../../../utils/userRoles'
+import { alterUser as alterUserReducer, setAlterUsers, setSelectedAlterUser } from '../../../store/reducers/adminQuery';
+import { getIndexOfK } from '../../../utils/helperFunctions'
+import { alterUser as alterUserDispatch } from '../../../store/reducers/adminQuery';
+
+const SelectedUserModal = ({showModal, closeModal, removeEntry=false, selectedAlterUser }) => {
+
+    const dispatch = useDispatch()
+    const {  traders, alterUsers, role } = useSelector(state => ({
+        alterUsers: state.adminQuery.alterUsers,
+        traders: state.adminQuery.traders,
+        role : state.user.role
+    }), shallowEqual)
+
+    const getTraders = [["Select Trader", ""], ...traders.map(trader => [`${trader.firstName} ${trader.lastName}`, trader.uid])]
+
+
+    const validationSchema = Yup.object().shape({
+        email: Yup.string().required().email().label("Email"),
+        firstName: Yup.string().required().min(3).label("First Name"),
+        lastName: Yup.string().required().min(3).label("Last Name"),
+        phone: Yup.string().required().label("Phone Number"),
+        roi: selectedAlterUser.role && selectedAlterUser.role.isUser && Yup.number().required().min(0).max(100).label("ROI"),
+        affiliateCode: selectedAlterUser.role && selectedAlterUser.role.isAffiliate && Yup.string().required().length(6).label("Affiliate Code"),
+        assignedTrader: selectedAlterUser.role && selectedAlterUser.role.isUser && Yup.string().required().label("Assigned Trader"),
+    })
+
+    const update = (newValue) => {
+        dispatch(setSelectedAlterUser({
+            ...selectedAlterUser,
+            role : JSON.parse(newValue)
+        }))
+    }
+
+    const alterUser = ({affiliateCode, roi, assignedTrader, role, affiliate}) => {
+        
+        dispatch(alterUserReducer({
+            ...selectedAlterUser,
+            roi, 
+            affiliateCode, 
+            assignedTrader,
+            role,
+            affiliate
+        }))
+
+        if(removeEntry) {
+            dispatch(setAlterUsers(
+                alterUsers.filter(user => user.uid !== selectedAlterUser.uid)
+            ))
+            dispatch(alterUserDispatch(selectedAlterUser))
```

Comment:
What is this for?

Reply:
This is the dispatch function that actually calls the generator that approves the user account.